File: monk_and_square_root.java
import java.io.*;
import java.util.*;
class HE
{
    public static void main(String[] args)
    {
        Scanner sc=new Scanner(System.in);
        int T;
        T=sc.nextInt();
        while(T-->0)
        {
            int N,M,ans=0,flag=1,i;
            N=sc.nextInt();
            M=sc.nextInt();
            for(i=0;;i++)
            {
                if((i*i)%M==N)
                {
                    
                    flag=i;
                    break;
                }
            }
            if(flag==0)
            {
                System.out.println("-1");
            }
            else
            {
                System.out.println(flag);
            }
        }
    }
}

File: monk_and_square_root.cpp
#include<iostream>
using namespace std;
int main()
{
    long long int T;
    cin>>T;
    while(T--)
    {
        long long int N,M,i,flag=0;
        cin>>N>>M;
        for(i=0;i<=m-1;i++)
        {
            if((i*i)%M==N)
            {
                flag=1;
                cout<<i<<endl;
                break;
            }
        }
        if(flag==0)
        {
            cout<<"-1"<<endl;
        }
    }
}
